### PR TITLE
Closes #682: Moderator toggle in participants list

### LINKED ISSUE #682: As a moderator I want to give moderator permissions to a participant via the participants list

### PR DESCRIPTION
**Pull request by brandstetterm**
## Description 

As a moderator I want to give moderator permissions to participants of the board via the participants list via a toggle.
The toggles are disabled for the own user and the board creator.

## Changelog

- Add owner/creator attribute to the board (to prevent removing the owner from the moderators list) 
- Add `changePermission` cloud function to modify the permissions of board participants as moderator
- Add redux workflow & api functions for modifying participant permissions
- Add `ToggleButton` component 
- Use `ToggleButton` in the participants list so that moderators are able to toggle participants permissions
- Update/add tests for all components
- Add SearchIcon svg

## Visual changes

Screenshot 2021-09-15 at 11 41 10
Screenshot 2021-09-15 at 11 41 01
Screenshot 2021-09-15 at 11 40 46
Screenshot 2021-09-15 at 11 40 31

### CODE REVIEW

File: src/api/user.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import {callAPI} from "./callApi";
+
+export const UserAPI = {
+  /**
+   * Changes the permissions of a participant.
+   *
+   * @param userId the identifier of the user whose permissions are being changed
+   * @param boardId the identifier of the board
+   * @param moderator the flag wheter the user recieves or loses moderator permissions
```
Comment (Dominik-Weinzierl) [inline src/api/user.ts:9]: Typo: whether

Comment (Dominik-Weinzierl) [inline src/api/user.ts:9]: Small typo: whether

**Review by Dominik-Weinzierl — COMMENTED** [review]
Looks good.

File: src/api/user.ts (hunk 1 above)
Comment (brandstetterm) [reply src/api/user.ts:9]: Thank you

**Review by Dominik-Weinzierl — APPROVED** [review]

**Pull request merged**